### PR TITLE
[FEAT] Optimize the make file with cmake

### PR DESCRIPTION
**Pull request by LeaYeh**
Learning cmake, and this is a simple version!

image

### CODE REVIEW

**Comment by LeaYeh** [comment]
@yenthing @leske42 merge this one first, it will be helpful

File: Makefile [hunk 1]
```diff
@@ -1,49 +1,49 @@
...
 all: $(NAME)
 
-$(NAME): $(OBJS) $(DEPS)
-	$(CXX) $(CXXFLAGS) $(INCFLAGS) $(LDFLAGS) $(OBJS) -o $(NAME)
+$(NAME): cmake_build
+	@$(MAKE) -C $(BUILD_DIR) $(MAKEFLAGS)
+	@echo ""
+	@echo ${STY_GRE}${STY_BOL} "✅ Build Complete" ${STY_RES}
+	@echo ${STY_BLU} "➜ Project: " ${STY_WHI}${PROJECT}${STY_RES}
+	@echo ${STY_BLU} "➜ Status: " ${STY_GRE} "Mandetory" ${STY_RES}
```
Comment (itislu) [inline Makefile:19]: Typo - should be `Mandatory`

File: Makefile [hunk 2]
```diff
@@ -1,49 +1,49 @@
...
+	@echo $(STY_GRE)"[INFO] Clean the build directory"$(STY_RES)
 
 fclean: clean
-	rm -f $(NAME)
+	@echo $(STY_GRE)"[INFO] Perform full clean"$(STY_RES)
 
 re: fclean
-	$(MAKE) all
+	@sleep 1
+	@$(MAKE) all
```
[the remark this answers is not in the document]
Comment (LeaYeh) [reply Makefile:43]: I will keep it at the moment, for some reason on my macos combine clean & -j will occur problem

[the remark this answers is not in the document]
Comment (itislu) [reply Makefile:43]: I will only believe that once I see it with my own eyes 👀

**Review by yenthing — APPROVED** [review]
crazy makefile

**Review by yenthing — APPROVED** [review]

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (LeaYeh) [reply Makefile:43]: I will show you

**Pull request merged**